js/file_provider/file_provider.onedrive.js

```diff
@@ -0,0 +1,183 @@
+    }
+
+    _ensureAccessTokenAcquired() {
+        if(this._accessTokenPromise) {
+            return this._accessTokenPromise;
+        }
+
+        const deferred = new Deferred();
+
+        if(this._accessToken) {
+            deferred.resolve();
+        } else {
+            ajax.sendRequest({
+                url: this._getAccessTokenUrl,
+                dataType: "json"
+            }).done(response => {
+                this._accessToken = response.token;
+                this._accessTokenPromise = null;
+                deferred.resolve();
+            });
+        }
+
+        this._accessTokenPromise = deferred.promise();
+        return this._accessTokenPromise;
+    }
+
+    _getEntriesByPath(path) {
+        const itemPath = this._prepareItemRelativePath(path);
+        const queryString = "?$select=" + REQUIRED_ITEM_FIELDS + "&$expand=children($select=" + REQUIRED_ITEM_FIELDS + ")";
+        const url = APP_ROOT_URL + itemPath + queryString;
+        return ajax.sendRequest({
+            url: url,
+            dataType: "json",
+            cache: false,
+            headers: { "Authorization": "Bearer " + this._accessToken }
+        });
+    }
+
+    _uploadFileChunk(uploadUrl, chunkBlob, chunkSize, uploadedSize, totalSize) {
+        const chunkEndPosition = uploadedSize + chunkSize - 1;
+        const contentRange = `bytes ${uploadedSize}-${chunkEndPosition}/${totalSize}`;
+        return ajax.sendRequest({
+            url: uploadUrl,
+            method: "PUT",
+            dataType: "json",
+            data: chunkBlob,
+            upload: {
+                onprogress: noop,
+                onloadstart: noop,
+                onabort: noop
+            },
+            cache: false,
+            headers: {
+                "Authorization": "Bearer " + this._accessToken,
+                "Content-Range": contentRange
+            }
+        });
+    }
+
+    _initiateUploadSession(fileId) {
+        const url = `${DRIVE_API_URL}/items/${fileId}/createUploadSession`;
+        return ajax.sendRequest({
+            url: url,
+            method: "POST",
+            dataType: "json",
+            cache: false,
+            headers: { "Authorization": "Bearer " + this._accessToken }
+        });
+    }
+
+    _createFile(folderPath, objectName) {
+        const itemPath = this._prepareItemRelativePath(folderPath);
+        const queryString = "?$select=" + REQUIRED_ITEM_FIELDS;
+        const url = APP_ROOT_URL + itemPath + "/children" + queryString;
+
+        const params = {
+            "name": objectName,
+            "file": { },
+            "@microsoft.graph.conflictBehavior": "rename"
+        };
+        const data = JSON.stringify(params);
+
+        return ajax.sendRequest({
+            url: url,
+            method: "POST",
+            dataType: "json",
+            data: data,
+            cache: false,
+            headers: {
+                "Authorization": "Bearer " + this._accessToken,
```

Comment:
I see several identical lines. Perhaps better to use getter?

```suggestion
                "Authorization": this._authorizationString,
```

```js
get _authorizationString() {
  return `Bearer ${this._accessToken}`;
}
```